### PR TITLE
Fixes #10701 - foreman-debug config file support

### PR DESCRIPTION
**Pull request by lzap**
This patch adds configuration file reading for foreman-debug. It allows to
redefine default values for all configuration options plus to configure some
advanced stuff like words to filter out or to temporarily disable upload
feature (e.g. for security policy reasons).

### CODE REVIEW

File: script/foreman-debug [hunk 1]
```diff
@@ -89,11 +92,7 @@ add_files() {
   done
 }
 
-# when running via sosreport close STDIN - http://projects.theforeman.org/issues/4200
-if ps -p $PPID | grep -q sosreport; then
-  exec 0</dev/null
-fi
```
Comment (lzap) [inline script/foreman-debug:95]: Note this is no longer required as interactive mode was removed.

**Comment by domcleal** [comment]
Can you please ship the config file?  I dislike ghost files, we're unlikely to notice bugs if it breaks.

File: script/foreman-debug [hunk 2]
```diff
@@ -119,6 +119,9 @@ else
   EXTENSION=""
 fi
 
+# read optional configuration file with user-defined defaults
+test -f /etc/foreman/foreman-debug.conf && source /etc/foreman/foreman-debug.conf
```
Comment (domcleal) [inline script/foreman-debug]: config/ in the Foreman root, traditionally, not /etc itself.

**Comment by ohadlevy** [comment]
@lzap bump?

**Comment by lzap** [comment]
Good points, amended.

**Comment by lzap** [comment]
This does not need any packaging update I assume. Since all values are commented out (sane defaults), I don't think we need sample-file approach.

**Comment by domcleal** [comment]
Sure it would, to be marked as a config file and installed into /etc/.

File: config/foreman-debug.conf [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+# Quiet mode (0 or 1)
+#QUIET=0
+
+# Verbose mode (0 or 1)
+#VERBOSE=0
+
+# Debug mode (0 or 1)
+#DEBUG=0
+
+# Upload tarbal after each run (0 or 1)
```
Comment (domcleal) [inline config/foreman-debug.conf]: tarball

**Comment by lzap** [comment]
Fixed, I see failures which seem to be unrelated.

http://ci.theforeman.org/job/test_develop_pr_core/5185/database=postgresql,ruby=2.1/testReport/junit/(root)/HostTest__create%20new%20host%20page/test_0002_default_primary_interface_is_in_the_overview_table/

**Comment by dLobatog** [comment]
Merged as cb5f02b3688e675ea68508c99823c136d55a154a, thanks @lzap!

**Comment by domcleal** [comment]
@lzap could you send in the packaging updates too please?  Install the config file as usual under /etc, lots of examples already.

**Comment by lzap** [comment]
Hmmm we don't read from `/etc/` path, only `/usr/share/foreman/config/foreman-debug.conf`. Should I only mark this file as a config file?

**Comment by lzap** [comment]
Oh a symlink for the rescue!

**Pull request closed without merging**